File: demos_code_J1_conception_objet/01_encapsulation_invariants/DemoEncapsulationAvant.java
import java.util.ArrayList;
import java.util.List;

/**
 * Version volontairement mauvaise : objet anémique et état modifiable librement.
 * À montrer rapidement avant la version protégée.
 */
public class DemoEncapsulationAvant {
    enum Statut { BROUILLON, PAYEE, EXPEDIEE }

    static final class Commande {
        public final List<Integer> prixCents = new ArrayList<>();
        public Statut statut = Statut.BROUILLON;
    }

    public static void main(String[] args) {
        Commande c = new Commande();
        c.prixCents.add(1200);
        c.prixCents.add(-500);       // incohérent, mais autorisé
        c.statut = Statut.EXPEDIEE;  // expédition sans paiement

        System.out.println("Prix = " + c.prixCents);
        System.out.println("Statut = " + c.statut);
        System.out.println("=> Le compilateur accepte un état métier incohérent.");
    }
}

File: Demo en Python/demos_code_J1_conception_objet_python/01_encapsulation_invariants/DemoEncapsulationAvant.py
from enum import Enum, auto


class Statut(Enum):
    BROUILLON = auto()
    PAYEE = auto()
    EXPEDIEE = auto()


class Commande:
    """Version volontairement mauvaise : état métier modifiable librement."""

    def __init__(self) -> None:
        self.prix_cents: list[int] = []
        self.statut = Statut.BROUILLON


def main() -> None:
    c = Commande()
    c.prix_cents.append(1200)
    c.prix_cents.append(-500)       # incohérent, mais autorisé
    c.statut = Statut.EXPEDIEE      # expédition sans paiement

    print(f"Prix = {c.prix_cents}")
    print(f"Statut = {c.statut.name}")
    print("=> Python accepte lui aussi cet état métier incohérent.")


if __name__ == "__main__":
    main()
